File: data/src/main/java/home/oleg/placesnearme/model/PhotoDbEntity.java
package home.oleg.placesnearme.model;

import android.arch.persistence.room.ColumnInfo;
import android.arch.persistence.room.Entity;
import android.arch.persistence.room.ForeignKey;
import android.arch.persistence.room.PrimaryKey;
import android.support.annotation.NonNull;

import static android.arch.persistence.room.ForeignKey.CASCADE;

@Entity(foreignKeys = @ForeignKey(entity = DetailedVenueDbEntity.class,
        parentColumns = "id",
        childColumns = "venueId",
        onDelete = CASCADE))
public class PhotoDbEntity {

    @PrimaryKey(autoGenerate = true)
    private Long id;
    private Long height;
    private String prefix;
    private String suffix;
    private Long width;
    @ColumnInfo(index = true)
    private String venueId;

    public Long getId() {
        return id;
    }

    public void setId(Long id) {
        this.id = id;
    }

    public String getVenueId() {
        return venueId;
    }

    public void setVenueId(String venueId) {
        this.venueId = venueId;
    }

    public Long getHeight() {
        return height;
    }

    public void setHeight(Long height) {
        this.height = height;
    }

    public String getPrefix() {
        return prefix;
    }

    public void setPrefix(String prefix) {
        this.prefix = prefix;
    }

    public String getSuffix() {
        return suffix;
    }

    public void setSuffix(String suffix) {
        this.suffix = suffix;
    }

    public Long getWidth() {
        return width;
    }

    public void setWidth(Long width) {
        this.width = width;
    }
}

File: core-data/src/main/java/home/oleg/placesnearme/coredata/model/PhotoDbEntity.kt
package home.oleg.placesnearme.coredata.model


import androidx.room.ColumnInfo
import androidx.room.Embedded
import androidx.room.Entity
import androidx.room.ForeignKey
import androidx.room.ForeignKey.CASCADE
import androidx.room.PrimaryKey
import home.oleg.placesnearme.coredomain.models.Photo

@Entity(foreignKeys = [ForeignKey(entity = DetailedVenueDbEntity::class,
        parentColumns = arrayOf("id"),
        childColumns = arrayOf("venueId"),
        onDelete = CASCADE)])
data class PhotoDbEntity(
        @PrimaryKey(autoGenerate = true)
        var id: Long = 0,
        @ColumnInfo(index = true)
        var venueId: String = "",
        @Embedded
        var photo: Photo)

